pkg/operator/ceph/cluster/controller.go

```diff
@@ -759,14 +760,22 @@ func (c *ClusterController) onDelete(obj interface{}) {
 }
 
 func (c *ClusterController) handleDelete(cluster *cephv1.CephCluster, retryInterval time.Duration) error {
-
+	flexDriverEnabled := os.Getenv(enableFlexDriver) == "true"
 	operatorNamespace := os.Getenv(k8sutil.PodNamespaceEnvVar)
+
+	if !flexDriverEnabled {
+		logger.Debugf("Flex driver disabled: no volume attachments for cluster %s operator namespace %s",
+			cluster.Namespace, operatorNamespace)
+		return nil
+	}
+
 	retryCount := 0
 	for {
 		// TODO: filter this List operation by cluster namespace on the server side
 		vols, err := c.volumeAttachment.List(operatorNamespace)
 		if err != nil {
-			return fmt.Errorf("failed to get volume attachments for operator namespace %s: %+v", operatorNamespace, err)
+			return fmt.Errorf("failed to get volume attachments for cluster %s operator namespace %s: %+v",
+				cluster.Namespace, operatorNamespace, err)
```

Comment:
nit: the statement doesn't quite make sense.

Reply:
I'll just revert it to the earlier message.